scripts/dts/gen_legacy_defines.py

```diff
@@ -746,6 +746,9 @@ def out(ident, val, aliases=(), deprecation_msg=None):
 
     for alias in aliases:
         if alias != ident:
+            if alias.startswith("INST_"):
+                deprecation_msg = "Macro is deprecated"
```

Comment:
Are you sure nobody is using these with `#ifdef`? As I mentioned in https://github.com/zephyrproject-rtos/zephyr/pull/24400, that won't work.

Reply:
We'll from my grep it showed no one obvious.  I figured just by introducing the PR we'll find some cases (and have).

After merged, this will get a nightly run so we can see if there are any other cases that got missed.

Reply:
OK